mbed_lstools/lstools_linux_generic.py

```diff
@@ -88,10 +88,10 @@ def list_mbeds(self):
         self.ERRORLEVEL_FLAG = 0
 
         result = []
-        tidhex = re.compile(r'_([0-9a-fA-F]+)-\d+:\d+')
+        tidpattern = re.compile(r'_([0-9a-zA-Z]+)-\d+:\d+')
```

Comment:
This is illegal here. We are searching for target id which is specified as hex ascii string. For example:
```
usb-MBED_microcontroller_02400226489a1e6c000000000000000000000000b543e3d4-0:0 -> ../../sdb
```
has a valid ```02400226489a1e6c000000000000000000000000b543e3d4``` Target ID.

In your case ```ATML2127031800007973``` is illegal target id. In any case, does your device have a ```mbed.htm``` file on MSD of interface chip?